### PR TITLE
[16.0] fs_attachment: sent the mimetype information for S3

### PR DESCRIPTION
**Pull request by sebastienbeau**
When storing image on amazon s3, ovh or scaleway.
The minetype must be set in order to have the right mimetype when requesting the image publicly.

Note : when using the api put_file it will guess the mimetype automatically : https://github.com/fsspec/s3fs/blob/47ffcfad33f8c74974a1515cb266168ede6cf4b5/s3fs/core.py#L1176

### CODE REVIEW

**Comment by vincent-hatakeyama** [comment]
Cherry picked in 17.0 to test with a local minio, mimetype is now correctly set.

File: fs_attachment/models/ir_attachment.py [hunk 1]
```diff
@@ -381,6 +383,13 @@ def _storage_file_read(self, fname: str) -> bytes | None:
             )
         return b""
 
+    def _storage_write_option(self, fs):
+        if hasattr(fs.fs, "s3"):
+            # For S3 storage the mimetype should be pass in the kwargs
+            return {"ContentType": self._context["mimetype"]}
```
Comment (lmignon) [inline fs_attachment/models/ir_attachment.py:389]: the s3 fs could be nested at a deeper level... IMO you should recursively test the nested fs. Something like:

```
_fs = fs
while _fs:
    if hasattr(_fs, "s3"):
        return {"ContentType": self._context["mimetype"]}
    _fs = getattr(_fs, "fs", None)
```

**Review by lmignon — CHANGES_REQUESTED** [review]

**Comment by lmignon** [comment]
superseded by #378

**Comment by sbidoul** [comment]
heads-up, in #497 this is moving to a separate fs_attachment_s3 module

**Pull request closed without merging**